modules/jupyter_service_accounts/service_accounts.tf

```diff
@@ -23,28 +23,40 @@ resource "google_service_account_iam_binding" "sa-binding" {
   role               = "roles/iam.workloadIdentityUser"
 
   members = [
-    "serviceAccount:${var.project_id}.svc.id.goog[${var.namespace}/default]",
+    "serviceAccount:${var.project_id}.svc.id.goog[${var.namespace}/${var.k8s_service_account}]",
   ]
 }
 
 resource "google_project_iam_binding" "project-binding" {
   project = var.project_id
-  for_each = toset(var.sa_iam_roles)
+  for_each = toset(var.gcp_sa_iam_roles)
   role = each.key
 
   members = [
     "serviceAccount:${google_service_account.sa.account_id}@${var.project_id}.iam.gserviceaccount.com",
   ]
 }
 
+resource "kubernetes_service_account" "ksa" {
```

Comment:
I got an error that kubernetes_service_account usage with the service account token automount is deprecated after 1.24 k8s - did you also see this? May need to use kubernetes_manifest